WordPress/Classes/Models/Blog/Blog+WordPressComRestAPI.swift

```diff
@@ -11,7 +11,8 @@ public extension Blog {
     /// - Warning: this method doesn't know if a Jetpack blog has the JSON API disabled.
     @objc
     var wordPressComRestApi: WordPressComRestApi? {
-        account?.wordPressComRestApi
+        // FIXME: We are banking on the fact that by the time a consumer reads this, WPAccount will already have initialized its API client instance.
+        account?._private_wordPressComRestApi
```

Comment:
Maybe we could decouple `WPAccount` `wordPressComRestApi` from WordPressAuthenticator using... a `Notification`? 

https://github.com/wordpress-mobile/WordPress-iOS/blob/b14619bc0426c23af1400a32bc470a416099e19e/WordPress/Classes/Models/WPAccount%2BRestApi.swift#L12-L20

https://github.com/wordpress-mobile/WordPress-iOS/blob/b14619bc0426c23af1400a32bc470a416099e19e/WordPress/Classes/Models/WPAccount%2BRestApi.swift#L27-L38

I suggest a `Notification` because I don't think injecting callbacks or using a delegate would be feasible with a model like this.

Reply:
See https://github.com/wordpress-mobile/WordPress-iOS/pull/24392